Source/Libs/Shared/GLibrary.cs

```diff
@@ -28,7 +28,7 @@ static GLibrary()
         _libraryDefinitions[Library.Gtk] = new[] { "libgtk-3-0.dll", "libgtk-3.so.0", "libgtk-3.0.dylib", "gtk-3.dll" };
         _libraryDefinitions[Library.PangoCairo] = new[] { "libpangocairo-1.0-0.dll", "libpangocairo-1.0.so.0", "libpangocairo-1.0.0.dylib", "pangocairo-1.dll" };
         _libraryDefinitions[Library.GtkSource] = new[] { "libgtksourceview-4-0.dll", "libgtksourceview-4.so.0", "libgtksourceview-4.0.dylib", "gtksourceview-4.dll" };
-        _libraryDefinitions[Library.Webkit] = new[] { "libwebkitgtk-3.0.dll", "libwebkitgtk-3.0.so.0", "libwebkitgtk-3.0.dylib", "libwebkitgtk-3.0.dll" };
+        _libraryDefinitions[Library.Webkit] = new[] { "libwebkit2gtk-4.0.dll", "libwebkit2gtk-4.0.so.0", "libwebkit2gtk-4.0.dylib", "libwebkit2gtk-4.0.0.dll" };
```

Comment:
On my debian stretch based linux it is "libwebkit2gtk-4.0.so.37" and same used in Eto.Forms.

Reply:
libwebkit2gtk-4.0.so.0 is a link to current subversion libwebkit2gtk-4.0.so.xx, which is in turn a link to a current subversion libwebkit2gtk-4.0.so.xx.yy.z

```
ls -l /usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so
lrwxrwxrwx 1 root root 23 Apr 16 12:59 /usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so -> libwebkit2gtk-4.0.so.37
```

```
ls -l /usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so.37
lrwxrwxrwx 1 root root 28 Apr 16 12:59 /usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so.37 -> libwebkit2gtk-4.0.so.37.53.1
```

Reply:
Usually ".0" files comes from "-dev" packages and not available to usual users.

Reply:
Check what package is installed that file.

Reply:
all the defs are like this:

`        _libraryDefinitions[Library.GLib] = new[] { "libglib-2.0-0.dll", "libglib-2.0.so.0", "libglib-2.0.0.dylib", "glib-2.dll" };`

```
ls -l /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 -> libglib-2.0.so.0.6400.6
```

Reply:
Same for ".so" without version in end.

Reply:
we could even cut the .0 at the end, i guess:

```
ls -l /usr/lib/x86_64-linux-gnu/libglib-2.0.so
/usr/lib/x86_64-linux-gnu/libglib-2.0.so -> libglib-2.0.so.0
```

but this would be another story ...

Reply:
> we could even cut the .0 at the end, i guess:

if you are the opinion, that 

> Usually ".0" files comes from "-dev" packages and not available to usual users.

we **should** cut the .0 at the end

Reply:
Different packages - different naming. Yes, glib main package (not -dev) has:
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0.5000.3

But libwebkit2gtk-4.0-37 has:
/usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so.37
/usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so.37.24.9
And libwebkit2gtk-4.0-dev has:
/usr/lib/x86_64-linux-gnu/libwebkit2gtk-4.0.so

As you see I don't have ".0" version at all (depends on linux distrib I guess).

Reply:
so cut the .0?

Reply:
removed it: https://github.com/GtkSharp/GtkSharp/pull/271/commits/96ad592ac102f5ce05973e9ee21421ade8e0abd8

Reply:
Use `libwebkit2gtk-4.0.so.37` as in eto.forms

Reply:
let's check that if we have a working example, testing it on different distros.

Reply:
Just some more info about so versions and native linking:
https://unix.stackexchange.com/a/476
quote:
> As you can see, it points to e.g. libpthread.so.0, not just libpthread.so.
> The reason for the symbolic link is for the linker.

Thats why "mylib.so" is comes from "-dev" packages.

Reply:
debian 9: libwebkit2gtk-4.0.so.37 , but no libwebkit2gtk-4.0.so

```
ls -l /usr/lib/x86_64-linux-gnu/libwebkit
libwebkit2gtk-4.0.so.37       libwebkitgtk-3.0.so.0
libwebkit2gtk-4.0.so.37.24.9  libwebkitgtk-3.0.so.0.22.17
```
centos 8: same story

```
ls /usr/lib64/libwebkit2gtk*.*
/usr/lib64/libwebkit2gtk-4.0.so.37  /usr/lib64/libwebkit2gtk-4.0.so.37.37.6
```

Reply:
> Use libwebkit2gtk-4.0.so.37 as in eto.forms

updated in: https://github.com/GtkSharp/GtkSharp/pull/271/commits/f6c69f8632771dd235e8f0bf7e0d41f645e6a80c